central/reports/scheduler/v2/reportgenerator/report_gen_impl.go

```diff
@@ -81,6 +82,30 @@ type ImageCVEInterface interface {
 	GetCveBaseInfo() *storage.CVEInfo
 }
 
+// buildDeployedImagesQueryParts builds query parts with pagination
+func buildDeployedImagesQueryParts(limit int32, offset int32) *ReportQueryParts {
```

Comment:
How about setting the limit at the top where `deployedImagesQueryParts` and `watchedImagesQueryParts` are declared so you don't have to add all these extra functions?